src/main/java/org/springframework/data/redis/support/collections/RedisList.java

```diff
@@ -229,4 +229,39 @@ default E moveLastTo(RedisList<E> destination, Direction destinationPosition, Du
 	 * @see <a href="https://redis.io/commands/ltrim">Redis Documentation: LTRIM</a>
 	 */
 	RedisList<E> trim(long start, long end);
+
+	@Override
+	default void addFirst(E e) {
+		List.super.addFirst(e);
+	}
+
+	@Override
+	default void addLast(E e) {
+		List.super.addLast(e);
+	}
+
+	@Override
+	default E getFirst() {
+		return List.super.getFirst();
+	}
+
+	@Override
+	default E getLast() {
+		return List.super.getLast();
+	}
+
+	@Override
+	default E removeFirst() {
+		return List.super.removeFirst();
+	}
+
+	@Override
+	default E removeLast() {
+		return List.super.removeLast();
+	}
+
+	@Override
+	default RedisList<E> reversed() {
+		throw new UnsupportedOperationException("Not Implemented");
```

Comment:
Let's turn that into a proper implementation using a package-private delegation type. Users looking at the code and finding `UOE` will have a hard time understanding why this is and whether `reversed` is supported at all.

Reply:
Well, methods  [ `addFirst(:E)`, 'addLast(:E)`, `getFirst()`, `getLast()`, `removeFirst(:E)`, `removeLast(:E)` ] must be redefined in `RedisList<E>` since `RedisList`, unlike Java's LinkedList (a "class"), is an "interface.

And, since the declarations/definitions of these methods differ between Java's `List` interface and the `Deque` interface, where `List` provides default implementations, and `Deque` does not, leads to the following compilation error:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (java-compile) on project spring-data-redis: Compilation failure
[ERROR] /Users/jblum/pivdev/spring-data-redis/src/main/java/org/springframework/data/redis/support/collections/RedisList.java:[40,7] error: types List<E#1> and BlockingDeque<E#2> are incompatible;
[ERROR]   interface RedisList<E#3> inherits abstract and default for addFirst(E#3) from types List and BlockingDeque
[ERROR]   where E#1,E#2,E#3 are type-variables:
[ERROR]     E#1 extends Object declared in interface List
[ERROR]     E#2 extends Object declared in interface BlockingDeque
[ERROR]     E#3 extends Object declared in interface RedisList
```

Reply:
`reversed()`on the other hand, can indeed be encapsulated in a proper type with a supporting implementation.